src/probes/index.ts

```diff
@@ -0,0 +1,33 @@
+import * as joi from '@hapi/joi';
+import 'joi-extract-type';
+
+import { NoopProbe, noopProbeSchema } from './NoopProbe';
+import { KafkaConsumerLagProbe, kafkaConsumerLagProbeSchema } from './KafkaConsumerLagProbe';
+import { Probe } from './Probe';
+
+const probeSchema = joi.alternatives().try(
+    noopProbeSchema.required(),
+    kafkaConsumerLagProbeSchema.required(),
+);
+
+export type ProbeConfig = joi.extractType<typeof probeSchema>;
+
+function buildProbe(probeConfig: ProbeConfig): Probe {
+    const type = probeConfig.type;
+    switch (type) {
+    case 'noop':
+        return new NoopProbe(probeConfig);
+    case 'kafkaConsumerLag':
+        return new KafkaConsumerLagProbe(probeConfig);
+    }
+
+    // not really dead code for transpiled uses
+    const tpe: string = probeConfig['type'];
+    throw new Error(`unsupported probe type '${tpe}'`);
```

Comment:
Returning a `noop` probe would basically be a silent failure, which is generally a no-no. The worst case scenario would be a probe configured with a typo in the name, acting as a noop, then allowing the system to be overloaded a week later just because the typo wasn't caught when people were paying attention.

In this case I just prefer to crash and make things very obvious before any real activity has taken place.

Reply:
Fair, thank you for the explanation!